Sources/ShopifyCheckoutKit/AnalyticsEvents.swift

```diff
@@ -0,0 +1,3579 @@
+        return Collection(
+            id: id ?? self.id,
+            productVariants: productVariants ?? self.productVariants,
+            title: title ?? self.title
+        )
+    }
+
+    func jsonData() throws -> Data {
+        return try newJSONEncoder().encode(self)
+    }
+
+    func jsonString(encoding: String.Encoding = .utf8) throws -> String? {
+        return String(data: try self.jsonData(), encoding: encoding)
+    }
+}
+
+/// The name of the customer event
+enum PixelEventsCollectionViewedName: String, Codable {
+    case collectionViewed = "collection_viewed"
+}
+
+/// The `page_viewed` event logs an instance where a customer visited a page.
+/// This event is available on the online store, checkout, and order status pages
+///
+/// The `page_viewed` event logs an instance where a customer visited a page.
+/// This event is available on the online store, checkout, and order status
+/// pages
+// MARK: - PixelEventsPageViewed
+struct PixelEventsPageViewed: Codable {
+    /// The client-side ID of the customer, provided by Shopify
+    let context: Context?
+    let data: PixelEventsPageViewedData?
+    /// The ID of the customer event
+    let id: String?
+    /// The name of the customer event
+    let name: PixelEventsPageViewedName?
+    /// The timestamp of when the customer event occurred, in [ISO
+    /// 8601](https://en.wikipedia.org/wiki/ISO_8601) format
+    let timestamp: String?
+
+    enum CodingKeys: String, CodingKey {
+        case context, data, id, name, timestamp
+    }
+}
+
+// MARK: PixelEventsPageViewed convenience initializers and mutators
+
+extension PixelEventsPageViewed {
+    init(data: Data) throws {
+        self = try newJSONDecoder().decode(PixelEventsPageViewed.self, from: data)
+    }
+
+    init(_ json: String, using encoding: String.Encoding = .utf8) throws {
+        guard let data = json.data(using: encoding) else {
+            throw NSError(domain: "JSONDecoding", code: 0, userInfo: nil)
+        }
+        try self.init(data: data)
+    }
+
+    init(fromURL url: URL) throws {
+        try self.init(data: try Data(contentsOf: url))
+    }
+
+    func with(
+        context: Context?? = nil,
+        data: PixelEventsPageViewedData?? = nil,
+        id: String?? = nil,
+        name: PixelEventsPageViewedName?? = nil,
+        timestamp: String?? = nil
+    ) -> PixelEventsPageViewed {
+        return PixelEventsPageViewed(
+            context: context ?? self.context,
+            data: data ?? self.data,
+            id: id ?? self.id,
+            name: name ?? self.name,
+            timestamp: timestamp ?? self.timestamp
+        )
+    }
+
+    func jsonData() throws -> Data {
+        return try newJSONEncoder().encode(self)
+    }
+
+    func jsonString(encoding: String.Encoding = .utf8) throws -> String? {
+        return String(data: try self.jsonData(), encoding: encoding)
+    }
+}
+
+// MARK: - PixelEventsPageViewedData
+struct PixelEventsPageViewedData: Codable {
```

Comment:
This one isn't typed very well in web pixels manager. Not sure if there's much we can do about it?